### PR TITLE
Indirect Sqw - Check Q and E are within the plot range

### LINKED ISSUE #26209: Indirect Sqw - Check if Q and E range is within correct boundaries before running
### Expected behavior
Check the entered Q and E range and show useful message if the range is not within the boundaries seen on the contour plot.

### Actual behavior
No check on Q and E. It will process fine, but nonsense data will be outputted if nonsense input is given.

### Steps to reproduce the behavior
1. `Interfaces`->`Indirect`->`Data Reduction`->`Sqw` tab
2. Load red file
3. Enter a Q and E which is not within the boundaries on the contour plot.
4. Click `Run`. There is no validation check on thee Q and E ranges.

### Platforms affected
All

### PR DESCRIPTION
**Pull request by robertapplin**
~~Merge #25956 First~~
----------------------

**Description of work.**
This PR adds better validation checks for data when it is loaded in to the Data Reduction interface.

**To test:**
1. `Interfaces`->`Indirect`->`Data Reduction`->`Sqw` tab
2. Load data below
3. Change the Q Low value to be 0.2. (This is out of the acceptable range)
4. Click `Run`. A useful message should appear.

**Data Files**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/3363322/irs26176_graphite002_red.zip)

Fixes #26209 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by LamarMoore — CHANGES_REQUESTED** [review]
The solution seems to work but there is a change mentioned below which should fix the issue with the other platforms.

File: qt/widgets/common/inc/MantidQtWidgets/Common/UserInputValidator.h [hunk 1]
```diff
@@ -81,9 +101,45 @@ class DLLExport UserInputValidator {
...
+
+/**
+ * Gets a workspace from the ADS.
+ *
+ * @param workspaceName The name of the workspace
+ * @return The workspace
+ */
+template <typename T, typename R>
+R UserInputValidator::getADSWorkspace(std::string const &workspaceName) {
+  return AnalysisDataService::Instance().retrieveWS<T>(workspaceName);
```
Comment (LamarMoore) [inline qt/widgets/common/inc/MantidQtWidgets/Common/UserInputValidator.h]: This needs to be fully qualified I think.

Comment (LamarMoore) [reply qt/widgets/common/inc/MantidQtWidgets/Common/UserInputValidator.h]: I applied the fix. @mantidproject/gatekeepers  will need to review this now.

**Review by LamarMoore — DISMISSED** [review]

**Comment by NickDraper** [comment]
@robertapplin Is this PR still "In Progress"?

**Comment by robertapplin** [comment]
Yes, there are a couple additions that need to be made. These are nearly done and I'll remove the label soon

**Comment by LamarMoore** [comment]
@robertapplin  There are semicolons at the end of function definitions in `IndirectDataValidationHelper.cpp` lines 107,127,149,88 which are causing the builds to fail.

**Comment by robertapplin** [comment]
@LamarMoore Thanks, I'll fix this now

**Comment by robertapplin** [comment]
I just need to finish writing the Unit tests for the IndirectDataValidationHelper, and then this will be ready to go.

**Comment by LamarMoore** [comment]
@robertapplin Linux is not happy about your `MockDataSelector` being in an anonymous namespace.

**Comment by robertapplin** [comment]
Thanks @LamarMoore , this should be fixed now.

**Comment by LamarMoore** [comment]
@mantidproject/gatekeepers this is ready again.

**Pull request merged**